src/Raven.Server/Documents/Indexes/Workers/MapItems.cs

```diff
@@ -138,12 +139,21 @@ protected MapItems(Index index, IndexStorage indexStorage, MapReduceIndexingCont
                                     try
                                     {
                                         current.KnownToBeNew = _indexStorage.LowerThanLastDatabaseEtagOnIndexCreation(current.Etag);
-                                        var numberOfResults = _index.HandleMap(current, mapResults,
-                                            writeOperation, indexContext, collectionStats);
 
-                                        resultsCount += numberOfResults;
-                                        collectionStats.RecordMapSuccess();
-                                        _index.MapsPerSec?.MarkSingleThreaded(numberOfResults);
+                                        if (itemEnumerator.Current.ShouldBeProcessedAsArchived(_index.ArchivedDataProcessingBehavior) == false)
+                                        {
+                                            var numberOfResults = _index.HandleMap(current, mapResults,
+                                                writeOperation, indexContext, collectionStats);
+                                            
+                                            resultsCount += numberOfResults;
+                                            
+                                            collectionStats.RecordMapSuccess();
+                                            _index.MapsPerSec?.MarkSingleThreaded(numberOfResults);
+                                        }
+                                        else
+                                        {   // skip from indexing
+                                            _index.HandleArchived(current, collection, writeOperation, indexContext, stats, itemEnumerator.Current.LowerId);
```

Comment:
Didn't we talk that `DeleteArchived()` would be more meaningful since under the covers it does deletion of archived documents

Reply:
done